qiskit/dagcircuit/dagcircuit.py

```diff
@@ -580,6 +580,22 @@ def _bits_in_operation(operation):
             else:
                 yield from node_resources(target).clbits
 
+    @staticmethod
+    def _operation_may_have_bits(operation) -> bool:
+        """Return whether a given :class:`.Operation` may contain any :class:`.Clbit` instances
+        in itself (e.g. a control-flow operation).
+
+        Args:
+            operation (qiskit.circuit.Operation): the operation to check.
+        """
+        # This is separate to `_bits_in_operation` because most of the time there won't be any bits,
+        # so we want a fast path to be able to skip creating and testing a generator for emptiness.
+        #
+        # If updating this, also update `DAGCirucit._bits_in_operation`.
+        return getattr(operation, "condition", None) is not None or isinstance(
+            operation, SwitchCaseOp
+        )
+
```

Comment:
This optimization is in addition to the optimization discussed in the opening comment, right? Did you benchmark this optimization?  It's clear that `set(self._bits_in_operation(op)).union(cargs)` will be slower than just `cargs` even when the generator is empty. But I would not be surprised if this is not measurable in benchmarks.